src/second.test.js

```diff
@@ -0,0 +1,29 @@
+import Html from './html-mock';
+import LocalStorage from './localstorage-mock';
+
+global.localStorage = new LocalStorage();
+
+const html = new Html();
+
+describe('DOM Manipulation', () => {
+  it('should add a item to the DOM', () => {
+    html.setInnerHTML(0, '<li>task1</li>');
+    expect(html.getInnerHtml(0)).toEqual('<li>task1</li>');
+  });
+});
+
+it('should remove an item from the dom', () => {
+  html.setInnerHTML(0, '<li>task1</li>');
+  html.setInnerHTML(1, '<li>task2</li>');
+  html.setInnerHTML(2, '<li>task3</li>');
+  html.removeInnerHtml(1);
+  expect(html.getInnerHtml(1)).toBeNull();
+});
+
+it('should edit the value of the input element', () => {
+  localStorage.setItem(1, 'item1');
+  localStorage.setItem(2, 'item2');
+  localStorage.setItem(3, 'item3');
+  localStorage.setItem(1, 'edit');
+  expect(localStorage.getItem(1)).toEqual('edit');
+});
```

Comment:
- [ ] These tests only prove that the mock localStorage that you created is working as expected. They have not really tested the app itself. We do testing to verify/assert that the functions written within the app are working properly. In this case, these tests do not touch on any part of the app and therefore are not proof that the app is working as expected.

  ### Points to take note of:
  - We don't test what we mock. Mocks are tools to use during testing, they are not part of the app's functionality.
  - Mocks' main purpose is to assist the testing process and prevent the functions under test from interacting with the real implementation of what we are testing because that can be expensive in terms of space and time.
  
  So, these tests for the mock localStorage are not really what we are after. The mock localStorage should only be used by the functions you are testing.

Reply:
> * We don't test what we mock. Mocks are tools to use during testing, they are not part of the app's functionality.
> * Mocks' main purpose is to assist the testing process and prevent the functions under test from interacting with the real implementation of what we are testing because that can be expensive in terms of space and time

you seem to have a good understanding of this test, I would be thankful if we make a call and discuss how to fix this.

Reply:
Hi,
No problem, I will reach out to you on slack.

Reply:
Hi @Lameck1 
I already sent you a direct message on slack.